powerline/segments/common/bat.py

```diff
@@ -9,6 +9,17 @@
 
 
 def _fetch_battery_info(pl):
+	def _flatten_battery_metadata(pl, batteries):
+		'''Flatten metadata from multiple batteries into a format that looks
+		like one battery in the end.'''
+		numerator = sum(map(lambda x: float(x[0]) * x[1], batteries))
+		energy_full_sum = sum(map(lambda x: x[1], batteries))
+		# We detect whether the computer is currently AC powered by checking
+		# if there are batteries which are not in the 'Charging' state.
+		ac_powered = any(map(lambda x: x[2], batteries))
+		energy_percentage = (numerator) / energy_full_sum
+		return lambda pl: (energy_percentage, ac_powered)
```

Comment:
It seems like that you do not understand that `_fetch_battery_info` is called only *once*. Returned function *must* fetch necessary values, `lambda pl: {constant}` may not be accepted.